src/library.js

```diff
@@ -3456,18 +3463,25 @@ LibraryManager.library = {
       } else {
         addr = lookup;
       }
-      if (addr.length >= nodelen) {
-        return {{{ cDefine('EAI_OVERFLOW') }}};
```

Comment:
Just highlighting some specific changes: here the old behavior was wrong that if there is an overflow, the string was not written at all, but an error was returned. The correct behavior is to write out the truncated string, but also return an overflowed result.